### PR TITLE
Security Bounded documentation

### PR DESCRIPTION
**Pull request by bukasaaime**
Security BC

1. Overview
2. User / Operator login (AuthN)
3. Authorization Model
4. BC bootstrap
5. BC startup
6. Role / Privilege association
7. Example Query / call

### CODE REVIEW

**Review by mdebarros — CHANGES_REQUESTED** [review]
A couple of comments.

Please make sure there are no spaces in the image names.

Also please be consistent with the file naming conventions. I see snake-case being used, and then another with spaces, and another with camel-case.

Note Camel-case is prefered.

File: docs/refarch/boundedContexts/security/index.md [hunk 1]
```diff
@@ -1,21 +1,71 @@
...
 
 ## Use Cases
 
-### Perform Transfer (universal mode)
+### User / Operator login (AuthN)
+
+The login function requires the user id and a secret key passed in the http body. The response contains a signed JWT token. The signature is generated by the Crypto subsystem.
+The login is performed by the Authorization services or IAM.
+
+![Use Case - Example REPLACE ME](./assets/securityBCv0.3.png)
```
Comment (mdebarros) [inline docs/refarch/boundedContexts/security/index.md:33]: Please change the info text for the image to something more useful then the default template `Use Case - Example REPLACE ME`.

File: docs/refarch/boundedContexts/security/index.md [hunk 2]
```diff
@@ -1,21 +1,71 @@
...
+
+### Authorization Model (AuthZ)
+
+IAM will provide users / groups, roles and privileges associations. Each BC will also have a list of related roles. When a function or microservice
+is called, the role provided in the JWT will be compared with the roles associated with the BC, after signature verfication. 
+
+![Use Case - Example REPLACE ME](./assets/securityBCv0.8.png)
```
Comment (mdebarros) [inline docs/refarch/boundedContexts/security/index.md:40]: Please change the info text for the image to something more useful then the default template `Use Case - Example REPLACE ME`.

File: docs/refarch/boundedContexts/security/index.md [hunk 3]
```diff
@@ -1,21 +1,71 @@
...
+
+### BC Bootstrap 
+
+At bootstrap, BC will send list of possible privileges. This is done once per deployement of a new version.
+
+![Use Case - Example REPLACE ME](./assets/securityBCv0.4.png)
```
Comment (mdebarros) [inline docs/refarch/boundedContexts/security/index.md:46]: Please change the info text for the image to something more useful then the default template `Use Case - Example REPLACE ME`.

File: docs/refarch/boundedContexts/security/index.md [hunk 4]
```diff
@@ -1,21 +1,71 @@
...
+
+### BC Startup 
+
+At startup the BC will request authentication issuer public keys from Security BC Crypto / KMS subsystems and the list of roles / privileges Security BC IAM subsystem. A local crypto library signature verification function will verfiy the JWT signature and the roles in the JWT will be compared with the local list of roles obtained from the central authorization service. 
+
+
+![Use Case - Example REPLACE ME](./assets/bc_start_up.png)
```
Comment (mdebarros) [inline docs/refarch/boundedContexts/security/index.md:53]: Please change the info text for the image to something more useful then the default template `Use Case - Example REPLACE ME`.

File: docs/refarch/boundedContexts/security/index.md [hunk 5]
```diff
@@ -1,21 +1,71 @@
...
+
+### Role / Privilege association 
+
+Roles are associated with a number of privileges.
+
+![Use Case - Example REPLACE ME](./assets/securityBCv0.6.png)
```
Comment (mdebarros) [inline docs/refarch/boundedContexts/security/index.md:59]: Please change the info text for the image to something more useful then the default template `Use Case - Example REPLACE ME`.

File: docs/refarch/boundedContexts/security/index.md [hunk 6]
```diff
@@ -1,21 +1,71 @@
...
+
+### Example Query / call 
+
+Client Authorization should be performed by using an access token. A client first needs to request the Authorization Service to create an access token for the user who requests to access the interface. The user is authenticated in the Authorization Service. The created access token is then used for authorization in the interface. 
+To use the access token, the client must set the Authorization HTTP header to Bearer [access_token] in each request to the interface.
+
+![Use Case - Example REPLACE ME](./assets/securityBCv0.7.png)
```
Comment (mdebarros) [inline docs/refarch/boundedContexts/security/index.md:66]: Please change the info text for the image to something more useful then the default template `Use Case - Example REPLACE ME`.

File: docs/refarch/boundedContexts/security/index.md [hunk 7]
```diff
@@ -1,21 +1,71 @@
...
 ## Terms
 
-Terms with specific and commonly accepted meaning within the Bounded Context in which they are used.
+Terms with specific and commonly accepted meaning within the Security Bounded Context.
 
-| Term | Description |
+| Crypto providers | Adapter that will provide cryptographic services and Key Management System |
 |---|---|
-| Term1 | Description1 |
+| IAM | Identity and Access Management. Adapter that will provide services for menus, profiles, roles and permissions.  |
```
Comment (mdebarros) [inline docs/refarch/boundedContexts/security/index.md:18]: I would suggest adding the following terms to this table:

- AuthN
- AuthZ
- JWT
- KMS

**Review by mdebarros — CHANGES_REQUESTED** [review]
I have added some comments.

Also, I recommend being consistent with the file naming conventions. I see snake-case, camel-case and one even with space for the image names. Our preference is camel-case, so please be consistent here.

**Review by pedrosousabarreto — APPROVED** [review]
Looks good for a first version, we can publish it now and tune it later.

**Review by mdebarros — APPROVED** [review]

**Comment by pedrosousabarreto** [comment]
Due to the structure change in main, these changes where cherry picked to a separate PR and have already being merged to main. 
Closing.

**Pull request closed without merging**